### ISSUE TITLE
Weird messages output on urxvt

### ISSUE THREAD

**Issue by alexkubica** [ryukinix/decorating#17]
I noticed that on urxvt terminal the output is messed up, see following [screencast](https://drive.google.com/open?id=0B9ehAG_Nj4CzWTRhb3ZoNENnem8).
Maybe it's a bug on urxvt?
I'm using archlinux distro and urxvt terminal.

**Comment by ryukinix** [comment]
We have the same bug using PowerShell with Windows. I think is something about the caracters of endline finishing: \r\n and \n.  This is defined here: https://github.com/ryukinix/decorating/blob/master/decorating/stream.py#L63

**Comment by alexkubica** [comment]
I guess it's https://github.com/ryukinix/decorating/blob/master/decorating/stream.py#L108.
Maybe '\r' deletes the new-line and therefore goes one line up.
How can I change and test it easily?
I was thinking of trying [this](http://stackoverflow.com/a/5291044).

**Comment by ryukinix** [comment]
@AlexK-IL You can do this:
1. clone the repo locally;
2. setup a virtualenv on root of repo:  "pyvenv env && source env/bin/activate"
3. call 'python setup.py develop' to install mal using symlink

Now the mal is installed on isolated environment and you can modify the source and test calling "env/bin/mal". That's a pretty easy way.

**Comment by ryukinix** [comment]
@ryukinix Wow, this thread is on decorating, sorry. I messed up. You can do the same about installation and try some usage case of @animated decorator on readme. I suggest to try first replicate the bug with a simple usage and fix the based on this simple case.

**Comment by alexkubica** [comment]
Thanks I'll try.

**Comment by alexkubica** [comment]
I'm a little stuck here.
I wrote a script to test the @animated decorator which I put in the same folder as `animation.py`.
It's content is:

```
#!/usr/bin/env python
# coding=utf-8

from decorating import animated
from time import sleep

@animated('test')
def main():
    sleep(3)


if __name__ == '__main__':
    main()
```

By running the script above I can reproduce the bug, however if I make changed to `animation.py` nothing changes.
The change I did was a simple one, I changed https://github.com/ryukinix/decorating/blob/master/decorating/animation.py#L207 so the message will contain something else then 'test'.
Maybe it's loading the installed decorating package instead of loading from local directory?
I don't know how to test it otherwise.

**Comment by ryukinix** [comment]
> Maybe it's loading the installed decorating package instead of loading from local directory?

Yes, is possible. The easilly way to avoid confusing about multiple versions is using virtual enviroments and MAKE SURE you are installed as symlink `python setup.py develop`. If you try install directly using `python setup.py install` this is a problem, because all the source is packaged and installed as it is in that moment, so any modification later doesn't have any effect.

Are you using virtual enviroments and symlinks for install?

Check `pip list` and send the output to here.

**Comment by alexkubica** [comment]
The problem with executing `python setup.py develop` is that I'm using a package manager to handle the files and it messes up with them, which leaves me with untracked files on my machine...
There should be a way to execute the scripts without installing them.

**Comment by ryukinix** [comment]
This is exactly the reason of the suggestion about use a virtual environment. You get the package installed without fuckup you system (yes, I'm agreed about install things without package manager is pretty bad too). Using venv all installed files will be in just in a simple directory that you can delete later.

Some python packages is really hard execute without installing because the package data model of python. In the most cases, when I'm trying figure out how this works in another packages, this is really a big problem.

However, you did execute the package without installing, right? Ok. The python docs makes clear when two packages, one local and installed, has the same names the priority is executing the local.

Are you considering your modification just don't have any effect? You can try modify really hard the code to break up and see if the REAL executed package is the local `decorating`.

**Comment by ryukinix** [comment]
BTW, I recommend use [`autoenv`](https://aur.archlinux.org/packages/autoenv-git/) too. Using virtualenv and autoenv is really simple. You just enter in the directory and the env automagically is set after setup. Actually, the `autoenv` just wrapps `cd` command to execute (when exists) a local script `.env`. In general my .env is just this:

``` bash
source env/bin/activate
```

**Comment by alexkubica** [comment]
As mentioned in https://github.com/ryukinix/mal/issues/17, running it like so works and shows changes too:

```
 PYTHONPATH=. python decorating/test.py
```

**Comment by alexkubica** [comment]
Created PR: https://github.com/ryukinix/decorating/pull/18, if you can review please :+1:

**Comment by ryukinix** [comment]
@AlexK-IL Thanks for find that trip! ❤️

**Comment by ryukinix** [comment]
Fixed after merge #18
